### PR TITLE
New checksum util to return bytes

### PR DESCRIPTION
**Pull request by orwaichman**
- [x] All [tests](https://github.com/jfrog/build-info-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/build-info-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: utils/checksum.go
Comment (orwaichman) [inline utils/checksum.go:1]: I have read the CLA Document and I hereby sign the CLA

Comment (orwaichman) [reply utils/checksum.go:1]: recheck

**Comment by orwaichman** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by orwaichman** [comment]
recheck

**Review by yahavi — APPROVED** [review]

**Pull request merged**